### ISSUE TITLE
[Feature Request]

### ISSUE THREAD

**Issue by 2025826903** [callous-youth/BOAT#25]
### Required prerequisites

- [x] I have searched the [Issue Tracker](https://github.com/callous-youth/BOAT/issues) to make sure this feature request has not already been submitted.
- [x] I have checked the documentation <https://boat.readthedocs.io/en/main/> to confirm this feature is not already supported.

### Motivation

I have been running the data hyper‑cleaning example with several operator combinations (e.g., `NGD+CG`, `DI,NGD+RAD,PTT`, `GDA,NGD` without NA) to compare their convergence behavior. The example prints metrics (accuracy, F1, val loss) per outer iteration, but there is no built‑in way to automatically collect these metrics and generate convergence plots. Currently, I must manually redirect logs, parse the output, and write a custom script to draw curves—this is repetitive and error‑prone, especially when exploring 5–10 combinations in a classroom or research setting.

### Proposed solution

Provide a lightweight utility—either as a method on `boat.Problem` (e.g., `problem.get_metric_history()`) or as a standalone helper in `boat_torch.utils`—that:

1. **Accumulates metrics** (loss, accuracy, etc.) during the optimization loop, optionally without requiring the user to manually capture `print` statements.
2. **Exports a simple plot** (e.g., using matplotlib) of the chosen metrics against outer iterations.
3. **Accepts multiple history objects** to overlay curves from different operator combinations on the same axes, making comparison immediate.
A minimal API sketch:
```python
# During training
history = b_optimizer.run_iter(..., return_metrics=True)

# After training
from boat_torch.visualization import plot_curves, compare_curves
plot_curves(history, metrics=["acc", "f1"], save_path="curve.png")
compare_curves([history1, history2], labels=["NGD+CG", "DI+RAD"], metrics=["acc"])

### Alternatives considered

_No response_

### Additional context

_No response_
